lib/experimental/class-wp-rest-navigation-controller.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Navigation_Controller class
+ *
+ * @package    Gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Base Templates REST API Controller.
+ */
+class WP_REST_Navigation_Controller extends WP_REST_Posts_Controller {
+
+	/**
+	 * Registers the controllers routes.
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+
+		parent::register_routes();
+
+		// Lists a single nav item based on the given id or slug.
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/fallbacks',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_fallbacks' ),
+					'permission_callback' => array( $this, 'get_fallbacks_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::READABLE ),
+				),
+				'allow_batch' => $this->allow_batch,
+				'schema'      => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	public function get_fallbacks_permissions_check( $request ) {
+		return true;
+	}
+
+
+
+	public function get_fallbacks() {
+
+		$should_skip = apply_filters( 'block_core_navigation_skip_fallback', false );
+
+		if ( $should_skip ) {
+			return array();
+		}
+
+		// Get the most recently published Navigation post.
+		$navigation_post = gutenberg_block_core_navigation_get_most_recently_published_navigation();
```

Comment:
Both `gutenberg_block_core_navigation_get_most_recently_published_navigation` and `gutenberg_block_core_navigation_maybe_use_classic_menu_fallback` are unrelated to the Nav block's render and therefore shouldn't be in the nav blocks `index.php` file.

We should extract them into a `WP_Navigation_Fallbacks` class or something to capture the domain in a single class.